File: services/web-profile-ssr/src/entities/Order/Order.ts
import type { Notification } from '../Notifications/Notifications';

export const ORDER_TYPE = {
  DORMITORY: 'dormitory',
  SCHOLARSHIP: 'scholarship',
  EDUCATION: 'education',
  GENERAL: 'general',
} as const;

export type OrderType = (typeof ORDER_TYPE)[keyof typeof ORDER_TYPE];

export type OrderAction = {
  title: string;
  action: string;
};

export type Order = {
  id: string;
  type: OrderType;
  number: string;
  title: string;
  date: string;
  additionalFields?: Record<string, string>;
  pdfUrl?: string;
  notifications?: Notification[];
  actions?: {
    primary?: OrderAction;
    secondary?: OrderAction;
  };
};

File: services/core-applications/app/schemas/order.py
from datetime import date
from typing import Any
from uuid import UUID

from pydantic import BaseModel


class OrderNotificationResponse(BaseModel):
    severity: str
    message: str
    action: str | None = None


class OrderResponse(BaseModel):
    id: UUID
    type: str
    number: str
    title: str
    date: date
    additional_fields: dict[str, Any] | None = None
    pdf_url: str | None = None
    actions: dict[str, Any] | None = None
    notifications_count: int = 0


class OrderDetailResponse(OrderResponse):
    notifications: list[OrderNotificationResponse] = []


class OrdersListResponse(BaseModel):
    orders: list[OrderResponse]
    total: int
    has_more: bool


class OrdersCountResponse(BaseModel):
    dormitory: int = 0
    scholarship: int = 0
    education: int = 0
    general: int = 0
